fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java

```diff
@@ -301,15 +257,9 @@ private void refreshTable(BaseTable updatedTable, long baseSnapshotId,
         IcebergTableName baseIcebergTableName = new IcebergTableName(dbName, tableName, baseSnapshotId);
         IcebergTableName updatedIcebergTableName = new IcebergTableName(dbName, tableName, updatedSnapshotId);
         long latestRefreshTime = tableLatestRefreshTime.computeIfAbsent(new IcebergTableName(dbName, tableName), ignore -> -1L);
-
-        List<String> updatedPartitionNames = updatedTable.spec().isPartitioned() ?
-                listPartitionNamesWithSnapshotId(updatedTable, dbName, tableName, updatedSnapshotId, executorService) :
-                new ArrayList<>();
-
         synchronized (this) {
-            partitionNames.put(updatedIcebergTableName, updatedPartitionNames);
             tables.put(updatedIcebergTableName, updatedTable);
-            partitionNames.invalidate(baseIcebergTableName);
+            partitionCache.invalidate(baseIcebergTableName);
```

Comment:
why not refresh partitionCache

Reply:
I think maybe it's easy just to invalidate cache.  And there is no need just to cache partition names only.

Reply:
Do you think it's okay

```
        partitionCache.invalidate(baseIcebergTableName);
        partitionCache.getUnchecked(updatedIcebergTableName);
        synchronized (this) {
            tables.put(updatedIcebergTableName, updatedTable);
        }
```